modules/oe_multilingual_url_suffix/src/Form/NegotiationUrlSuffixForm.php

```diff
@@ -0,0 +1,139 @@
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageManagerInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation\LanguageNegotiationUrlSuffix;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Configure the URL suffixes language negotiation method for this site.
+ *
+ * @package Drupal\oe_multilingual_url_suffix\Form
+ */
+class NegotiationUrlSuffixForm extends ConfigFormBase {
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManagerInterface
+   */
+  protected $languageManager;
+
+  /**
+   * Constructs a new NegotiationUrlSuffixForm object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The factory for configuration objects.
+   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
+   *   The language manager.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, LanguageManagerInterface $language_manager) {
+    parent::__construct($config_factory);
+    $this->languageManager = $language_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('language_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'language_negotiation_configure_url_suffix_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['oe_multilingual_url_suffix.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    global $base_url;
+    $config = $this->config('oe_multilingual_url_suffix.settings');
+
+    $form['suffix'] = [
+      '#type' => 'details',
+      '#tree' => TRUE,
+      '#title' => $this->t('Path suffix configuration'),
+      '#open' => TRUE,
+      '#description' => $this->t('Language codes or other custom text to use as a path suffix for URL language detection. For the selected fallback language, this value may be left blank. <strong>Modifying this value may break existing URLs. Use with caution in a production environment.</strong> Example: Specifying "deutsch" as the path suffix code for German results in URLs like "example.com/contact_deutsch".'),
+    ];
+
+    $languages = $this->languageManager->getLanguages();
+    $suffixes = $config->get('url_suffixes');
+    foreach ($languages as $langcode => $language) {
+      $t_args = ['%language' => $language->getName(), '%langcode' => $language->getId()];
+      $form['suffix'][$langcode] = [
+        '#type' => 'textfield',
+        '#title' => $language->isDefault() ? $this->t('%language (%langcode) path suffix (Default language)', $t_args) : $this->t('%language (%langcode) path suffix', $t_args),
```

Comment:
Can't we just append ` (Default language)` to the title if language is default? It will make it more readable.

Reply:
Changed